sections/layer2/deposit/DepositTab/DepositTab.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useRecoilValue } from 'recoil';
+
+import { TabContainer } from '../../components/common';
+import { Transaction } from 'constants/network';
+
+import { getGasEstimateForTransaction } from 'utils/transactions';
+import useStakingCalculations from 'sections/staking/hooks/useStakingCalculations';
+import synthetix from 'lib/synthetix';
+import Notify from 'containers/Notify';
+import { appReadyState } from 'store/app';
+import { walletAddressState } from 'store/wallet';
+
+import ApproveModal from 'components/ApproveModal';
+import TabContent from './TabContent';
+import { normalizedGasPrice } from 'utils/network';
+
+const DepositTab = () => {
+	const { t } = useTranslation();
+	const { transferableCollateral } = useStakingCalculations();
+	const walletAddress = useRecoilValue(walletAddressState);
+	const isAppReady = useRecoilValue(appReadyState);
+	const { monitorHash } = Notify.useContainer();
+
+	const [gasLimitEstimate, setGasLimitEstimate] = useState<number | null>(null);
+	const [depositTxError, setDepositTxError] = useState<string | null>(null);
+	const [gasEstimateError, setGasEstimateError] = useState<string | null>(null);
+	const [isApproved, setIsApproved] = useState<boolean>(false);
+	const [gasPrice, setGasPrice] = useState<number>(0);
+	const [transactionState, setTransactionState] = useState<Transaction>(Transaction.PRESUBMIT);
+	const [txModalOpen, setTxModalOpen] = useState<boolean>(false);
+	const [txHash, setTxHash] = useState<string | null>(null);
+
+	useEffect(() => {
+		const getGasLimitEstimate = async () => {
+			if (isAppReady && walletAddress && isApproved && transferableCollateral) {
+				try {
+					setGasEstimateError(null);
+					const {
+						contracts: { SynthetixBridgeToOptimism },
+						utils: { parseEther },
+					} = synthetix.js!;
+					const gasEstimate = await getGasEstimateForTransaction(
+						[parseEther(transferableCollateral.toString())],
+						SynthetixBridgeToOptimism.estimateGas.initiateDeposit
+					);
+					setGasLimitEstimate(gasEstimate);
+				} catch (e) {
+					console.log(e);
+					setGasEstimateError(e.message);
+				}
+			}
+		};
+		getGasLimitEstimate();
+		// eslint-disable-next-line
+	}, [walletAddress, isAppReady, isApproved, transferableCollateral]);
+
+	const getAllowance = useCallback(async () => {
```

Comment:
does this work? I thought you couldn't pass an async method to the useCallback. I always just define an async method in the callback and then call it after it is defined inside of a sync method